docs/design/integrate-with-cc-tools/README.md

```diff
@@ -0,0 +1,300 @@
+
+    ```go
+    package stubwrapper
+
+    import (
+        "github.com/hyperledger-labs/cc-tools/errors"
+        "github.com/hyperledger/fabric-chaincode-go/shim"
+    )
+
+    type StubWrapper struct {
+        Stub        shim.ChaincodeStubInterface
+        WriteSet    map[string][]byte
+        PvtWriteSet map[string]map[string][]byte
+    }
+
+    func (sw *StubWrapper) PutState(key string, obj []byte) errors.ICCError {
+        err := sw.Stub.PutState(key, obj)
+        if err != nil {
+            return errors.WrapError(err, "stub.PutState call error")
+        }
+
+        if sw.WriteSet == nil {
+            sw.WriteSet = make(map[string][]byte)
+        }
+        sw.WriteSet[key] = obj
+
+        return nil
+    }
+
+    ```
+
+    On the other hand for FPC, it also wraps the [shim.ChaincodeStub](https://github.com/hyperledger/fabric-chaincode-go/blob/acf92c9984733fb937fba943fbf7397d54368751/shim/interfaces.go#L28) interface from Hyperledger Fabric but the wrapper it uses (in this case it's [FpcStubInterface](https://github.com/hyperledger/fabric-private-chaincode/blob/33fd56faf886d88a5e5f9a7dba15d8d02d739e92/ecc_go/chaincode/enclave_go/shim.go#L17)) is not always using the functions from the passed stub. With the same example as before, if you look at the [PutState](https://github.com/hyperledger/fabric-private-chaincode/blob/33fd56faf886d88a5e5f9a7dba15d8d02d739e92/ecc_go/chaincode/enclave_go/shim.go#L104) function you can notice it's not using the `sw.Stub.PutState` and going directly to the `rwset.AddWrite` (this is specific to fpc use case as it's not using the fabric proposal response). There are some other functions where the passed `stub` functions are being used.
+
+    ```go
+    package enclave_go
+
+    import (
+        "github.com/hyperledger/fabric-chaincode-go/shim"
+        pb "github.com/hyperledger/fabric-protos-go/peer"
+    )
+
+    type FpcStubInterface struct {
+        stub  shim.ChaincodeStubInterface
+        input *pb.ChaincodeInput
+        rwset ReadWriteSet
+        sep   StateEncryptionFunctions
+    }
+
+    func (f *FpcStubInterface) PutState(key string, value []byte) error {
+        encValue, err := f.sep.EncryptState(value)
+        if err != nil {
+            return err
+        }
+        return f.PutPublicState(key, encValue)
+    }
+
+    func (f *FpcStubInterface) PutPublicState(key string, value []byte) error {
+        f.rwset.AddWrite(key, value)
+
+        // Note that since we are not using the fabric proposal response  we can skip the putState call
+        // return f.stub.PutState(key, value)
+        return nil
+    }
+    ```
+
+    This was also tested practically by logging during the invocation of a transaction that uses the PutState functionality.
+    ![1726531513506](chaincodeStubOrder.png)
+
+    For this, It's better to inject the fpc stub wrapper in the middle between fabric stub and cc-tools stub wrapper.
```

Comment:
another place where fpc is lower case ...

Reply:
Done